MfciPkg/Include/Library/MfciDeviceIdSupportLib.h

```diff
@@ -8,48 +8,45 @@
   SPDX-License-Identifier: BSD-2-Clause-Patent
 **/
 
-#ifndef __MFCI_DEVICE_ID_SUPPORT_LIB_H__
-#define __MFCI_DEVICE_ID_SUPPORT_LIB_H__
+#ifndef MFCI_DEVICE_ID_SUPPORT_LIB_H_
+#define MFCI_DEVICE_ID_SUPPORT_LIB_H_
 
 #include <MfciVariables.h>
 
-// define a generic function prototype shared by all library functions
-
 /**
- * Get a device-unique targeting value
- *
- * @param[out] String   Device targeting value, a UTF-16 little endian string.
- *                      Includes a wide NULL terminator.
- *                      Refer to MfciPkg/Include/MfciVariables.h for more details.
- *
- * @param[out] StringSize  (OPTIONAL) String size in bytes including the wide NULL terminator.
- *                         NULL may be supplied if the size is not requested (it is NULL terminated after all)
- *
- * It is the callers responsibility to free the String buffer returned using FreePool()
- *
- * @return EFI_STATUS
- *
- * @retval EFI_UNSUPPORTED         Likely using the NULL library instance
- * @retval EFI_SUCCESS             Successfully retrieved the string and length
- */
-
+  Function pointer definition to get a device-unique targeting value.
+  It is the callers responsibility to free the String buffer returned using FreePool().
+
+  @param[out] String      Device targeting value, a UTF-16 little endian string.
+                          Includes a wide NULL terminator.
+                          Refer to MfciPkg/Include/MfciVariables.h for more details.
+  @param[out] StringSize  (OPTIONAL) String size in bytes including the wide NULL terminator.
+                          NULL may be supplied if the size is not requested (it is NULL terminated after all)
+
+  @retval EFI_UNSUPPORTED       Likely using the NULL library instance.
+  @retval EFI_SUCCESS           Successfully retrieved the string and length.
+  @retval EFI_OUT_OF_RESOURCES  There is not enough memory to allocate the string.
+  @retval EFI_INVALID_PARAMETER The String is NULL.
+**/
 typedef
 EFI_STATUS
 (EFIAPI *MFCI_DEVICE_ID_FN)(

```

Comment:
Does this still need to be in the public library interface?

Reply:
Yes, unfortunately the global data is still referenced in some platform code.

Reply:
It would be nice to clean up the interface some more, but given the potential impact on platforms, I'll approve as-is.

Reply:
We should see if the platform can drop the dependency on these array items at some point. If so, we could easily move this into a module for better sanity.